tools/remove_duplicate_entries.py

```diff
@@ -13,47 +13,58 @@ def __init__(self):
                 description = "Removes superfluous duplicate entries from polkit profiles or warns about conflicting ones."
         )
 
+
     def run(self):
         self.m_args = self.m_parser.parse_args()
 
-        def checkDuplicate(entry):
-            seen = self.m_actions_seen.get(entry.action, None)
-            if not seen:
-                self.m_actions_seen[entry.action] = entry
-            else:
-                if entry.settings == seen.settings:
-                    self.m_lines_to_drop.add(entry.linenr)
-                else:
-                    printerr("{}:{}: {}: conflicting duplicate entry ({}), previously seen in line {} ({})".format(
-                        seen.path.name.rjust(35),
-                        str(entry.linenr).rjust(3),
-                        seen.action,
-                        ':'.join(entry.settings),
-                        seen.linenr,
-                        ':'.join(seen.settings)
-
-                    ))
-
         for profile in PROFILES:
 
             self.m_lines_to_drop = set()
             self.m_actions_seen = {}
 
             path = getProfilePath(profile)
-            parseProfile(path, checkDuplicate)
-            self.rewriteProfile(path, self.m_lines_to_drop)
+            for entry in parseProfile(path):
+                self.checkDuplicate(entry)
+
+            if self.m_lines_to_drop:
+                self.rewriteProfile(path, self.m_lines_to_drop)
+            else:
+                print("{}: no entries removed".format(path.name.ljust(35)))
+
+
+    def checkDuplicate(self, entry):
+        seen = self.m_actions_seen.get(entry.action, None)
+        if not seen:
+            self.m_actions_seen[entry.action] = entry
+        else:
+            if entry.settings == seen.settings:
+                self.m_lines_to_drop.add(entry.linenr)
+                print("{}:{}: removing redundant entry with same settings as in line {}".format(
+                    entry.path.name.ljust(35),
+                    str(entry.linenr).rjust(3),
+                    seen.linenr
+                ))
+            else:
+                printerr("{}:{}: {}: conflicting duplicate entry ({}), previously seen in line {} ({})".format(
+                    seen.path.name.ljust(35),
+                    str(entry.linenr).rjust(3),
+                    seen.action,
+                    ':'.join(entry.settings),
+                    seen.linenr,
+                    ':'.join(seen.settings)
+
+                ))
+
 
     def rewriteProfile(self, path, lines_to_drop):
 
         lines = []
 
         with open(path) as fd:
 
-            linenr = 0
-
-            for line in fd.readlines():
+            for nr, line in enumerate(fd.readlines()):
```

Comment:
sorry, should have been more clear: `enumerate(.., start=1)`

Reply:
Ah, didn't know about the `start` parameter. Done.